sdk/core/Azure.Core.TestFramework/src/ClientTestFixtureAttribute.cs

```diff
@@ -14,65 +14,104 @@ namespace Azure.Core.TestFramework
         private readonly object[] _serviceVersions;
         private readonly int? _maxServiceVersion;
 
+        /// <summary>
+        /// Initializes an instance of the <see cref="ClientTestFixtureAttribute"/> accepting additional fixture parameters.
+        /// </summary>
+        /// <param name="serviceVersions">The set of service versions that will be passed to the test suite.</param>
         public ClientTestFixtureAttribute(params object[] serviceVersions)
         {
             _serviceVersions = serviceVersions;
 
             _maxServiceVersion = _serviceVersions.Any() ? _serviceVersions.Max(s => Convert.ToInt32(s)) : (int?)null;
         }
 
+        /// <summary>
+        /// Initializes an instance of the <see cref="ClientTestFixtureAttribute"/> accepting additional fixture parameters.
+        /// </summary>
+        /// <param name="serviceVersions">The set of service versions that will be passed to the test suite.</param>
+        /// <param name="additionalParameters">An array of additional parameters that will be passed to the test suite.</param>
+        public ClientTestFixtureAttribute(object[] serviceVersions, object[] additionalParameters)
+        {
+            Assert.IsNotNull(additionalParameters, nameof(additionalParameters));
+
+            _additionalParameters = additionalParameters ?? new object[] { };
+            _serviceVersions = serviceVersions ?? new object[] { };
+
+            _maxServiceVersion = _serviceVersions.Any() ? _serviceVersions.Max(s => Convert.ToInt32(s)) : (int?)null;
+        }
+
         public IEnumerable<TestSuite> BuildFrom(ITypeInfo typeInfo)
         {
             return BuildFrom(typeInfo, this);
         }
 
         public IEnumerable<TestSuite> BuildFrom(ITypeInfo typeInfo, IPreFilter filter)
         {
+
+            var suitePermutations = GeneratePermutations();
+
+            foreach (var (fixture, isAsync, serviceVersion, parameter) in suitePermutations)
+            {
+                foreach (TestSuite testSuite in fixture.BuildFrom(typeInfo, filter))
+                {
+                    Process(testSuite, serviceVersion, isAsync, parameter);
+                    yield return testSuite;
+                }
+            }
+        }
+
+        private List<(TestFixtureAttribute suite, bool isAsync, object serviceVersion, object paramter)> GeneratePermutations()
+        {
+            var result = new List<(TestFixtureAttribute suite, bool isAsync, object serviceVersion, object paramter)>();
+
             if (_serviceVersions.Any())
             {
                 foreach (object serviceVersion in _serviceVersions)
                 {
-                    var syncFixture = new TestFixtureAttribute(false, serviceVersion);
-                    var asyncFixture = new TestFixtureAttribute(true, serviceVersion);
-
-                    foreach (TestSuite testSuite in asyncFixture.BuildFrom(typeInfo, filter))
+                    if (_additionalParameters?.Any() ?? false)
```

Comment:
Is it even null?

Reply:
Good catch - since refactoring, it won't be.